src/devices/bosch.ts

```diff
@@ -551,6 +551,79 @@ const tzLocal = {
             }
         },
     } satisfies Tz.Converter,
+    bosch_room_thermostat: {
```

Comment:
Could you convert this to modernExtend, some examples: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/sonoff.ts#L831

This replace all the toZigbee/fromZigbee and exposes code.

Reply:
> Has the cluster name in deviceAddCustomCluster to be unique (per device, globally)? 

Unique per device

>  Can modernExtend partially override/extend a default cluster with manufacturer-specific attributes? E.g.: 0x4007 in hvacThermostat for operating_mode?

Yes! Just use the existing cluster name for the `name` parameter.

Your PR looks good, can I merge it?

Reply:
Looks good, thanks!